app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,481 @@
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                resetScreen(reqError as Error);
+                return;
+            }
+
+            close();
+        } catch (e) {
+            resetScreen(e as Error);
+        }
+    }), [userInfo]);
+
+    const resetScreen = useCallback((resetError: Error) => {
+        setError(resetError?.message);
+        Keyboard.dismiss();
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    }, [enableSaveButton]);
+
+    const updateField = useCallback((fieldKey: string, name: string) => {
+        const update = {...userInfo};
+        update[fieldKey] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[fieldKey];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo, currentUser]);
+
+    const includesSsoService = (sso: string) => ['gitlab', 'google', 'office365'].includes(sso);
+    const isSAMLOrLDAP = (protocol: string) => ['ldap', 'saml'].includes(protocol);
+
+    const userProfileFields: FieldSequence = {
+        firstName: {
+            ref: firstNameRef,
+            isDisabled: (isSAMLOrLDAP(service) && lockedFirstName) || includesSsoService(service),
+        },
+        lastName: {
+            ref: lastNameRef,
+            isDisabled: (isSAMLOrLDAP(service) && lockedLastName) || includesSsoService(service),
+        },
+        username: {
+            ref: usernameRef,
+            isDisabled: service !== '',
+        },
+        email: {
+            ref: emailRef,
+            isDisabled: true,
+        },
+        nickname: {
+            ref: nicknameRef,
+            isDisabled: isSAMLOrLDAP(service) && lockedNickname,
+        },
+        position: {
+            ref: positionRef,
+            isDisabled: isSAMLOrLDAP(service) && lockedPosition,
+        },
+    };
+
+    const hasDisabledFields = Object.values(userProfileFields).filter((field) => field.isDisabled);
+
+    const onFocusNextField = useCallback(((fieldKey: string) => {
+        const findNextField = () => {
+            const fields = Object.keys(userProfileFields);
+            const curIndex = fields.indexOf(fieldKey);
+            const searchIndex = curIndex + 1;
+
+            if (curIndex === -1 || searchIndex > fields.length) {
+                return undefined;
+            }
+
+            const remainingFields = fields.slice(searchIndex);
+
+            const nextFieldIndex = remainingFields.findIndex((f: string) => {
+                const field = userProfileFields[f];
+                return !field.isDisabled;
```

Comment:
This seems a "strong" dependency on the userProfileFields object. Up to this point you were only using the keys (which will not change) but isDisabled might change between renders. Am I right? Therefore, this should depend on userProfileFields. If that happens, we would probably want to memo userProfileFields...

Reply:
Refactored as this scenario can happen.

Reply:
Since userProfileFields is an object created on every render, just adding the dependence will make this to always be recalculated, and as a domino effect, anything else depending on this.
Either we memo the userProfileIds, or we remove this as a callback.

Reply:
Done.